qemu/tests/chardev_serial_login.py

```diff
@@ -0,0 +1,46 @@
+from virttest import error_context, env_process
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify the login function of pci-serial (RHEL and x86 only):
+    1) Start guest with pci-serial with backend unix_socket
+    2) append console=ttyS0 to guest kernel config
+    3) append /dev/ttyS0 to /etc/security
+    4) reboot guest
+    5) nc host/path for unix and tcp backend.
+    6) login to guest and move files, restart network
+    7) repeat step 1 to 6 with pty, tcp backend
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    serial_id = params.objects('serials')[-1]
+    params['start_vm'] = 'yes'
+    for backend in ['unix_socket', 'tcp_socket', 'pty']:
+        params['chardev_backend_%s' % serial_id] = backend
+        vm = params['main_vm']
+        env_process.preprocess_vm(test, params, env, vm)
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        session = vm.wait_for_login()
+        assert session.cmd('lspci | grep "PCI 16550A"') is not None
+        session.cmd('echo "console=tty0 console=ttyS0,115200" >> /boot/grub2/grub.conf')
```

Comment:
This command is incorrect, to update kernel command line, can use 'grub2-editenv - set ***'.

Reply:
And, usually, this parameter will be added during installation. skip this step is acceptable.